gst/nnstreamer/include/nnstreamer_plugin_api_filter.h

```diff
@@ -615,6 +615,84 @@ extern void
 nnstreamer_filter_shared_model_replace (void *instance, const char *key,
     void *new_interpreter, void (*replace_callback) (void *, void *), void (*free_callback) (void*));
 
+/**
+ * @brief GstTensorTrainer's properties for neural network framework (internal data structure)
+ *
+ * Internal data of GstTensorTrainer required by tensor_trainer's custom subplugin.
+ */
+
+typedef struct _GstTensorTrainerProperties
+{
+  const char *model_config;      /* The configuration file path for creating model */
+  unsigned int input_tensors_size[NNS_TENSOR_SIZE_LIMIT];       /* Size of tensor in byte for each input type */
+  unsigned int num_input_tensors;       /* The number of input tensors, The number of input tensors is equal to the number of inputs plus the number of labels. */
+  unsigned int num_inputs;      /* The number of input lists, the input is where framework receive the features to train the model, num_inputs indicates how many inputs there are. */
+  unsigned int num_labels;      /* The number of label lists, the label is where framework receive the class to train the model, num_labels indicates how many labels there are. */
+  unsigned int num_train_samples;       /* The number of train sample used to train the model. */
+  unsigned int num_valid_samples;       /* The number of valid sample used to train the model. */
+} GstTensorTrainerProperties;
```

Comment:
For the tensor metadata, use nnstreamer standard data structure: i.e., ```GstTensorsInfo```

Reply:
Thank you.
1. We can calculate the size of each tensors using GstTensorsInfo and gst_tensor_info_get_size(&info->info[index]).
Is it against policy to calculate size and pass it to sub-plugin?